pkg/execution/state/redis_state/v2_adapter.go

```diff
@@ -212,10 +214,13 @@ func (v v2) SaveStep(ctx context.Context, id state.ID, stepID string, data []byt
 
 	return util.WithRetry(
 		ctx,
+		"state.SaveStep",
 		func(ctx context.Context) (bool, error) {
 			return v.mgr.SaveResponse(ctx, v1id, stepID, string(data))
 		},
-		util.NewRetryConf(),
+		util.NewRetryConf(
+			util.WithRetryConfRetryableErrors(v.retryableError),
```

Comment:
only adding the filter in here for now since this is where the issue happens the most.